plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedPassageFormatter.java

```diff
@@ -120,11 +122,27 @@ public String toString() {
     static MarkupPassage mergeAnnotations(AnnotationToken[] annotations, Passage passage) {
         MarkupPassage markupPassage = new MarkupPassage();
 
+        final Integer[] matches = new Integer[passage.getNumMatches()];
```

Comment:
Can this just be an `int[]`?

Reply:
Hey @romseygeek. Thanks for your review. It cannot be a `int[]` because `Arrays.sort` with a comparator doesn't support primitive types.